### ISSUE TITLE
Daylight Savings and scheduled deployments

### ISSUE THREAD

**Issue by Dalmirog** [OctopusDeploy/Issues#1511]
A user reported that his deployment was moved forward by an hour after the DST change. Creating this ticket to open the investigation.

User ticket: http://help.octopusdeploy.com/discussions/problems/31234

**Comment by PaulStovell** [comment]
We're going to close this as wont-fix - it's subjective, but I don't believe there is a bug here, just proof that daylight savings (and any changes to time in general) are bad ideas and should not be allowed to be touched by politicians. Time zones are hard. 

Imagine that I live in Sydney, and at midnight the clocks will change by one hour, from GMT+10 to GMT+11. 

At 8:46pm the day before the clock change, I go to schedule a deployment for 3am the day after. Here's the UI we currently show:

![2015-11-12 20_46_38-create deployment - octopus deploy](https://cloud.githubusercontent.com/assets/47085/11116464/7f2e8eac-897e-11e5-8d06-0a31fd0a74aa.png)

At 3am the next day, I check the deployment, and it hasn't started yet - it won't start until 4am Sydney time. What?!

This can be explained by looking at that UI closely: 

 - We show the time zone offset clearly. If we showed "3am Sydney time", then I would agree it would be a bug. But we show "3am GMT+10", which is unambiguous. 3am GMT+10 will happen at the same time in Brisbane (which doesn't observe daylight savings) and Sydney (which does). It just so happens that at 8:46pm the day prior, Sydney was at GMT+10, and the next day Sydney is at GMT+11. 
 - We also show the estimate of *when* the deployment will happen - about 6 hours from now. This is correct. If we started a stopwatch at 9am the day before, and waited until 3am Sydney time, the stopwatch would only say 5 hours (since the clock jumped ahead by an hour). If we kept waiting until 4am Sydney time then the clock would say 6 hours. 

Keep in mind the constraints we're dealing with:

 - Localities and time zone offsets are not the same thing. Sydney/NSW/Australia observes daylight savings time. +10:00 is just an offset, which doesn't relate to daylight savings. We can't assume that just because the clock is GMT+10 it may change to GMT+11 tomorrow. 
 - The Octopus server is in a time zone, and the browser may be in another time zone
 - We know the GMT offset for both the Octopus server and browser. We don't know their location or whether daylight savings is imminent. I think some of this information is in the registry, and perhaps with a lot of effort we could try to figure it out to provide a warning, but that would only help on the Octopus server side, not the browser. 

For these reasons I'm going to close as wont-fix. I'm open to suggestions on whether there's something we could do to make it clearer, but we did originally spend quite a bit of time thinking about this when we built it.
